xgcm/grid.py

```diff
@@ -1120,6 +1121,18 @@ def __init__(
             all_axes = comodo.get_all_axes(ds)
             coords = {}
 
+        # check coords input validity
+        for axis, positions in coords.items():
+            for pos, dim in positions.items():
+                if dim not in ds.variables:
+                    raise ValueError(
+                        f"Could not find dimension `{dim}` (for the `{pos}` position on axis `{axis}`) in input dataset."
+                    )
```

Comment:
This is the problem. You are requiring the dim to be in `ds.variables`. But that's not the case for datasets with no coordinate variable, e.g.
```python
ds = xr.DataArray([0], dims='x', name='foo').to_dataset()
'x' in ds.variables  # --> False
```